src/Components/Forms/src/FieldIdentifier.cs

```diff
@@ -111,20 +111,32 @@ private static void ParseAccessor<T>(Expression<Func<T>> accessor, out object mo
 
             // Get a reference to the model object
             // i.e., given an value like "(something).MemberName", determine the runtime value of "(something)",
-            switch (memberExpression.Expression)
+            if (memberExpression.Expression is ConstantExpression constantExpression)
             {
-                case ConstantExpression constantExpression:
-                    model = constantExpression.Value;
-                    break;
-                default:
-                    // It would be great to cache this somehow, but it's unclear there's a reasonable way to do
-                    // so, given that it embeds captured values such as "this". We could consider special-casing
-                    // for "() => something.Member" and building a cache keyed by "something.GetType()" with values
-                    // of type Func<object, object> so we can cheaply map from "something" to "something.Member".
-                    var modelLambda = Expression.Lambda(memberExpression.Expression);
-                    var modelLambdaCompiled = (Func<object>)modelLambda.Compile();
-                    model = modelLambdaCompiled();
-                    break;
+                if (constantExpression.Value is null)
+                {
+                    throw new ArgumentException("The provided expression must evaluate to a non-null value.");
```

Comment:
I changed the implementation a bit here. Creating a `FieldIdentifier` with a null value will results in an argument null exception. This provides a slightly better error experience.